wolfcrypt/src/asn.c

```diff
@@ -16953,7 +16970,30 @@ static int DecodeCertExtensions(DecodedCert* cert)
             int length = dataASN[CERTEXTASN_IDX_VAL].length;
 
             /* Decode the extension by type. */
-            ret = DecodeExtensionType(input + idx, length, oid, critical, cert);
+            ret = DecodeExtensionType(input + idx, length, oid, critical, cert,
+                                      &unknown);
+#if defined(WOLFSSL_CUSTOM_OID) && defined(WOLFSSL_ASN_TEMPLATE) \
+    && defined(HAVE_OID_DECODING)
+            if (unknown && (cert->unknownExtCallback != NULL)) {
+                word16 decOid[16];
```

Comment:
Please use `byte`. We need a build-time macro to enable using larger size.
```
#ifndef WC_ASN_UNKNOWN_MAX_LEN
#define WC_ASN_UNKNOWN_MAX_LEN 32
#endif
```

Reply:
I got this from `DumpOID()`.  Should I change it there as well?

Reply:
Yes please

Reply:
OH...whoops....are you sure you want this? 

```
int DecodeObjectId(const byte* in, word32 inSz, word16* out, word32* outSz)
```

I use word16 because that what `DecodeObjectId()` outputs.

Reply:
```
 * @param [in]      in     Byte array containing OID.
 * @param [in]      inSz   Size of OID in bytes.
 * @param [in]      out    Array to hold dotted form of OID.
 * @param [in, out] outSz  On in, number of elemnts in array.
 *                         On out, count of numbers in dotted form.
```